libs/langchain/langchain/chat_models/fake.py

```diff
@@ -16,28 +16,77 @@
 class FakeMessagesListChatModel(BaseChatModel):
     """Fake ChatModel for testing purposes."""
 
-    responses: List[BaseMessage]
+    responses: Union[List[BaseMessage], List[List[BaseMessage]]]
     sleep: Optional[float] = None
     i: int = 0
 
+    @property
+    def _llm_type(self) -> str:
+        return "fake-messages-list-chat-model"
+
+    @property
+    def _identifying_params(self) -> Dict[str, Any]:
+        return {"responses": self.responses}
+
     def _generate(
         self,
         messages: List[BaseMessage],
         stop: Optional[List[str]] = None,
         run_manager: Optional[CallbackManagerForLLMRun] = None,
         **kwargs: Any,
     ) -> ChatResult:
+        response = self._call(messages, stop=stop, run_manager=run_manager, **kwargs)
+        generation = ChatGeneration(message=response)
```

Comment:
I suspect this may have weird behavior if response is a list of chunks

Reply:
I don't think so, I'm using this in one of my tests and it works fine:

```python
    function_responses = [
        [
            AIMessageChunk(
                content="",
                additional_kwargs={"function_call": {"name": "test_function", "arguments": ""}},
            ),
            AIMessageChunk(
                content="",
                additional_kwargs={
                    "function_call": {"arguments": '{\n  "word": "foo",\n  "repeats": 2\n}'}
                },
            ),
        ],
        [
            AIMessageChunk(content="Foo repeated twice is: "),
            AIMessageChunk(content="foo foo"),
        ],
    ]
```